### ISSUE TITLE
Implement action to Download Universal APK from Google Play Console

### ISSUE THREAD

**Issue by AliSoftware** [wordpress-mobile/release-toolkit#485]
This is the ruby code that adds that functionality to fastlane's `Supply::Client` client:

```ruby
require 'supply'

# Maybe at some point we will submit a PR to fastlane to add this to the main features? (would need to wrap it into an action first)
module Supply
    ApkDownloadID = Struct.new(:package_name, :version_code, :download_id)

    class Client
        # @param [Fixnum] version_code
        #   Version code of the app bundle.
        # @return [Array<ApkDownloadID>]
        #
        def get_univeral_apk_list(package_name:, version_code:)
            result = call_google_api { client.list_generatedapks(package_name, version_code) }
        
            result.generated_apks.map do |row|
                ApkDownloadID.new(package_name, version_code, row.generated_universal_apk.download_id)
            end
        end

        # @param [Supply::ApkDownloadID] apk_id
        #   The ApkDownloadID retrieved from `get_univeral_apk_list`
        # @param [IO, String] destination
        #   IO stream or filename to receive content download
        #
        def download_univeral_apk(apk_id:, destination:)
            call_google_api {
                client.download_generatedapk(apk_id.package_name, apk_id.version_code, apk_id.download_id, download_dest: destination)
            }
        end
    end
end
```

Example usage for Tumblr Android:
```ruby
PLAY_STORE_JSON_KEY = File.join('.configure-files', 'push2play-key.json')
PACKAGE_NAME = 'com.tumblr'
version_code = 1290600109

client = File.open(PLAY_STORE_JSON_KEY) do |io|
    Supply::Client.new(service_account_json: io, params: { timeout: 5 })
end
universal_apks = client.get_univeral_apk_list(package_name: PACKAGE_NAME, version_code: version_code)
universal_apks.each do |apk|
    dest = File.join(Dir.home, 'Downloads', "#{apk.package_name}-universal-#{apk.version_code}.apk")
    puts "Downloading #{dest}..."
    client.download_univeral_apk(apk_id: apk, destination: dest)
    puts "Done."
end
```

We need to turn that code into a Fastlane action that would:
 - Take a `package_name`, `version_code`, `json_key` and `destination` input parameters (`ConfigItem`s)
 - Call the code above, improving it to add error handling, and auto-retry (especially in case we call this too soon after having uploaded the corresponding AAB if GPC haven't had time to generate the APK from it just yet), etc.

p1684504716992959-slack-C02KLTL3MKM

**Comment by AliSoftware** [comment]
I've now [opened a PR in the fastlane repo](https://github.com/fastlane/fastlane/pull/21315) to propose that new action to be included directly in fastlane itself 🎉

**Comment by AliSoftware** [comment]
My new `download_apk_from_google_play` action has [been shipped in `fastlane 2.214.0`](https://github.com/fastlane/fastlane/releases/tag/2.214.0); we can thus close this issue.
